krux_boto/__init__.py

```diff
@@ -33,6 +32,21 @@
 NAME            = 'krux-boto'
 DEFAULT_REGION  = 'us-east-1'
 
+
+def get_instance_region():
+    """
+    Query the instance metadata service and return the region this instance is
+    placed in. If the metadata service can't be contacted, return a generic
+    default instead.
+    """
+    ### TODO: XXX This shouldn't get called if we're not on EC2.
+    zone = boto.utils.get_instance_metadata().get('placement', {}).get('availability-zone', None)
+    if zone is None:
+        krux.logging.get_logger('krux_boto').warn('get_instance_region failed to get the local instance region')
+        return 'us-east-1'
```

Comment:
This was originally in kelb.py and called there to get the region and use it for the initial connection. It then proceeded to ignore this and reconnect to all the regions anyway, so it wasn't really in active use.

I thought that it could be useful to be able to get the region of the server we're running on for some scripts. I'm not using this anywhere currently, though, so I can just remove this if you'd rather. Another option would be to make it raise an Exception if it can't find the region, if that's what you mean by violating the POLA.